app/src/main/java/net/sourceforge/opencamera/OpenCameraApplication.java

```diff
@@ -27,4 +39,30 @@ private void checkAppReplacingState() {
             Process.killProcess(Process.myPid());
         }
     }
+
+    private class Core {
```

Comment:
To separate file please, and this class have to be initialized with main context and provide maybe some dependencies - dont call his init methods from app directly